### PR TITLE
handle different startup commands for maven and gradle

### LINKED ISSUE #5091: Handle different StartupCommand for maven and gradle
Currently `startupCommand` in ModuleBuilder is used to specificy maven wrapper command (`/mvnw`).

However once Gradle java build tool will be finished and available, generated documentation will not be accurate, since it will reference maven command for projects using gradle.

This will require a bit a refactoring and re-engineering since modules don't have knowledge of previously applied modules (here maven or gradle).

### PR DESCRIPTION
**Pull request by renanfranca**
Fix #5091 

Needed by:
- https://github.com/jhipster/jhipster-lite/pull/8867

### CODE REVIEW

File: src/main/java/tech/jhipster/lite/module/domain/JHipsterModule.java [hunk 1]
```diff
@@ -520,6 +530,18 @@ public JHipsterModuleBuilder startupCommand(String startupCommand) {
       return this;
     }
 
+    public JHipsterModuleBuilder startupMavenCommand(String startupCommand) {
```
Comment (murdos) [inline src/main/java/tech/jhipster/lite/module/domain/JHipsterModule.java:533]: Seems OK.
However we should fully switch existing startup commands to this new way of handling startup commands.
E.g. MavenModuleFactory should use `startupMavenCommand()`.

And I think existing `moduleBuilder.startupCommand(String)` could be renamed to `moduleBuilder.startupDockerCommand(String)` and be associated to startup command with type StartupCommandType.DOCKER_COMPOSE.
Thus tech.jhipster.lite.module.domain.JHipsterModuleShortcuts#startupCommand could be removed

File: src/main/java/tech/jhipster/lite/module/domain/file/startupcommand/JHipsterModuleStartupCommands.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
+package tech.jhipster.lite.module.domain.file.startupcommand;
```
Comment (murdos) [inline src/main/java/tech/jhipster/lite/module/domain/file/startupcommand/JHipsterModuleStartupCommands.java:1]: It should be just:
```suggestion
package tech.jhipster.lite.module.domain.startupcommand;
```

**Review by murdos — APPROVED** [review]

File: src/test/java/tech/jhipster/lite/module/infrastructure/secondary/FileSystemStartupCommandsReadmeCommandsHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+  }
+
+  @Test
+  void shouldNotAddMavenCommandToMavenProjectWithoutReadme() {
+    JHipsterProjectFolder projectFolder = projectFrom("src/test/resources/projects/" + EMPTY_MAVEN);
+    JHipsterStartupCommand command = createCommand("./mvnw clean verify sonar:sonar", StartupCommandType.MAVEN);
+
+    handler.handle(projectFolder, new JHipsterStartupCommands(List.of(command)));
+
+    // assertThat(content(projectFolder.filePath("README.md"))).doesNotContain("./mvnw clean verify sonar:sonar");
```
[the remark this answers is not in the document]
Comment (renanfranca) [reply src/test/java/tech/jhipster/lite/module/infrastructure/secondary/FileSystemStartupCommandsReadmeCommandsHandlerTest.java]: Thanks for the review!

**Comment by renanfranca** [comment]
@pascalgrimaud: Bounty Claimed https://opencollective.com/generator-jhipster/expenses/189215. Thanks 😄 ✌️

**Comment by pascalgrimaud** [comment]
@renanfranca : approved

**Pull request merged**